cljfmt/src/cljfmt/core.cljc

```diff
@@ -341,6 +346,35 @@
   ([form indents alias-map]
    (transform form edit-all should-indent? #(indent-line % indents alias-map))))
 
+(defn map-key? [zloc]
+  (and
+    (z/map? (z/up zloc))
+    (even? (index-of zloc))
+    (not (uneval-node? zloc))
+    (not (whitespace-or-comment? zloc))))
+
+(defn preceded-by-line-break? [zloc]
+  (loop [previous (zip/left zloc)]
```

Comment:
This could be written more concisely as:

```clojure
  (loop [previous (zip/left zloc)]
    (if (whitespace-or-comment? previous)
      (recur (zip/left previous))
      (and previous (line-break? previous))))
```